doc/api/deprecations.md

```diff
@@ -0,0 +1,574 @@
+# Deprecated APIs
+
+Node.js may deprecate APIs when either: (a) use of the API is considered to be
+unsafe, (b) an improved alternative API has been made available, or (c)
+breaking changes to the API are expected in a future major release.
+
+Node.js utilizes two kinds of Deprecations:
+
+* Documentation-only
+* Runtime
+
+A Documentation-only deprecation is one that is expressed only within the
+Node.js API docs. These generate no side-effects while running a Node.js
+process.
+
+A Runtime deprecation will, by default, generate a process warning that will
+be printed to `stderr` the first time the deprecated API is used within a
+Node.js process. When the `--throw-deprecation` command-line flag is used,
+a Runtime deprecation will cause an error to be thrown.
+
+## Un-deprecation
+
+From time-to-time the deprecation of an API may be reversed. Such action may
+happen in either a semver-minor or semver-major release. In such situations,
+this document will be updated with information relevant to the decision.
+*However, the deprecation identifier will not be modified*.
+
+## List of Deprecated APIs
+
+<a id="DEP0001"></a>
+### DEP0001: http.OutgoingMessage.prototype.flush
+
+Type: Runtime
+
+The `OutgoingMessage.prototype.flush()` method is deprecated. Use
+`OutgoingMessage.prototype.flushHeaders()` instead.
+
+<a id="DEP0002"></a>
+### DEP0002: require('\_linklist')
+
+Type: Runtime
+
+The `_linklist` module is deprecated. Please use a userland alternative.
```

Comment:
For this particular PR, I would rather not delve into the specifics of each individual deprecation description. I'd rather save that for subsequent PRs. Most of these I pulled directly from the warning messages printed to the console